Analysis/DataModel/include/Analysis/JetFinder.h

```diff
@@ -0,0 +1,234 @@
+                                                                                                               strategy(fastjet::Best),
+                                                                                                               areaType(fastjet::active_area),
+                                                                                                               algorithmBkg(fastjet::JetAlgorithm(fastjet::kt_algorithm)),
+                                                                                                               recombSchemeBkg(fastjet::RecombinationScheme(fastjet::E_scheme)),
+                                                                                                               strategyBkg(fastjet::Best),
+                                                                                                               areaTypeBkg(fastjet::active_area),
+                                                                                                               bkgSubMode(BkgSubMode::none),
+                                                                                                               constSubAlpha(1.0),
+                                                                                                               constSubRMax(0.6),
+                                                                                                               isReclustering(false)
+
+  {
+
+    //default constructor
+  }
+
+  /// Default destructor
+  ~JetFinder() = default;
+
+  /// Sets the jet finding parameters
+  void setParams()
+  {
+
+    if (!isReclustering) {
+      jetEtaMin = etaMin + jetR; //in aliphysics this was (-etaMax + 0.95*jetR)
+      jetEtaMax = etaMax - jetR;
+    }
+
+    //selGhosts =fastjet::SelectorRapRange(ghostEtaMin,ghostEtaMax) && fastjet::SelectorPhiRange(phiMin,phiMax);
+    //ghostAreaSpec=fastjet::GhostedAreaSpec(selGhosts,ghostRepeatN,ghostArea,gridScatter,ktScatter,ghostktMean);
+    ghostAreaSpec = fastjet::GhostedAreaSpec(ghostEtaMax, ghostRepeatN, ghostArea, gridScatter, ktScatter, ghostktMean); //the first argument is rapidity not pseudorapidity, to be checked
+    jetDef = fastjet::JetDefinition(algorithm, jetR, recombScheme, strategy);
+    areaDef = fastjet::AreaDefinition(areaType, ghostAreaSpec);
+    selJets = fastjet::SelectorPtRange(jetPtMin, jetPtMax) && fastjet::SelectorEtaRange(jetEtaMin, jetEtaMax) && fastjet::SelectorPhiRange(jetPhiMin, jetPhiMax);
+    jetDefBkg = fastjet::JetDefinition(algorithmBkg, jetBkgR, recombSchemeBkg, strategyBkg);
+    areaDefBkg = fastjet::AreaDefinition(areaTypeBkg, ghostAreaSpec);
+    selRho = fastjet::SelectorRapRange(bkgEtaMin, bkgEtaMax) && fastjet::SelectorPhiRange(bkgPhiMin, bkgPhiMax); //&& !fastjet::SelectorNHardest(2)    //here we have to put rap range, to be checked!
+  }
+
+  /// Sets the background subtraction estimater pointer
```

Comment:
I am not convinced this should be in the header. Instead, I'd prefer to have the implementation in a separate cxx file, which then is build and also pulls in the required libs (FastJet + contrib), which now needs to happen in every task including `JetFinder.h` (even if nothing of contrib is used by that task).

Do we want/need to support jet finding without FJcontrib?

Reply:
yes I also agree, i left it all in the header now as we decided to do a bit of reshuffling in the next stage but we can move everything to an appropriate .cxx
I think we can potentially support the jet finding without contrib for the cases were certain subtractions are not used if it makes a difference to compile time. although i guess this should be minimal

Reply:
The question is more if an installation of FastJet without contrib should be supported?

Reply:
Ah I see. In my opinion there isnt much need for this as anyone who decides to build with fastjet wont mind and perhaps in most cases will even require the contrib packages. We just might need some flexibility with managing different versions of fastjet and the contrib but i guess this should be ok to do